File: backend/lambdas/broadcast-lambda/src/main/java/com/artcorner/erp/dto/response/reports/EmployeePerformanceReportResponse.java
package com.artcorner.erp.dto.response.reports;

import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

@Data
@Builder
public class EmployeePerformanceReportResponse {

    private LocalDate fromDate;
    private LocalDate toDate;

    private List<UUID> employeeIds;
    private List<String> employeeNames;
    private List<String> employeeEmails;

    private List<Integer> employeeOrders;
    private List<Integer> employeeCompletedOrders;

    private List<BigDecimal> employeeRevenues;
    private List<BigDecimal> employeeProfits;
}

File: frontend/Admin/src/features/reports/types/EmployeePerformanceReportResponse.ts
export interface EmployeePerformanceReportResponse {
    fromDate: string;
    toDate: string;

    employeeIds: string[];
    employeeNames: string[];
    employeeEmails: string[];

    employeeOrders: number[];
    employeeCompletedOrders: number[];

    employeeRevenues: number[];
    employeeProfits: number[];
}